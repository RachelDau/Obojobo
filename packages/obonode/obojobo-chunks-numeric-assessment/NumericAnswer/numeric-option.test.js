import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import NumericOption from './numeric-option'

describe('NumericOption', () => {
	test('NumericOption renders as expected with requirement of `exact`', () => {
		const component = renderer.create(<NumericOption numericChoice={{ requirement: 'exact' }} />)

		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('NumericOption with requirement of `exact`, event.stopPropagation() is called when click', () => {
		const component = mount(<NumericOption numericChoice={{ requirement: 'exact' }} />)

		const stopPropagation = jest.fn()

		component
			.find('.select-item')
			.at(0)
			.simulate('click', {
				stopPropagation
			})

		component
			.find('.input-item')
			.at(0)
			.simulate('click', {
				stopPropagation
			})

		const tree = component.html()

		expect(tree).toMatchSnapshot()
		expect(stopPropagation).toBeCalledTimes(2)
	})

	test('NumericOption renders as expected with requirement of `range`', () => {
		const component = renderer.create(<NumericOption numericChoice={{ requirement: 'range' }} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('NumericOption with requirement of `range`, event.stopPropagation() is called when click', () => {
		const component = mount(<NumericOption numericChoice={{ requirement: 'range' }} />)

		const stopPropagation = jest.fn()
		component
			.find('.select-item')
			.at(0)
			.simulate('click', {
				stopPropagation
			})

		component
			.find('.input-item')
			.at(0)
			.simulate('click', {
				stopPropagation
			})

		component
			.find('.input-item')
			.at(1)
			.simulate('click', {
				stopPropagation
			})

		const tree = component.html()

		expect(tree).toMatchSnapshot()
		expect(stopPropagation).toBeCalledTimes(3)
	})

	test('NumericOption renders as expected with requirement of `margin`', () => {
		const component = renderer.create(<NumericOption numericChoice={{ requirement: 'margin' }} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('NumericOption with requirement of `margin`, event.stopPropagation() is called when click', () => {
		const component = mount(<NumericOption numericChoice={{ requirement: 'margin' }} />)

		const stopPropagation = jest.fn()

		component
			.find('.select-item')
			.at(0)
			.simulate('click', {
				stopPropagation
			})

		component
			.find('.select-item')
			.at(1)
			.simulate('click', {
				stopPropagation
			})

		component
			.find('.input-item')
			.at(0)
			.simulate('click', {
				stopPropagation
			})

		component
			.find('.input-item')
			.at(1)
			.simulate('click', {
				stopPropagation
			})
		const tree = component.html()

		expect(tree).toMatchSnapshot()
		expect(stopPropagation).toBeCalledTimes(4)
	})

	test('NumericOption with requirement of `margin`, percent error, event.stopPropagation() is called when click', () => {
		const component = mount(
			<NumericOption numericChoice={{ requirement: 'margin', type: 'percent' }} />
		)

		const stopPropagation = jest.fn()

		component
			.find('.select-item')
			.at(0)
			.simulate('click', {
				stopPropagation
			})

		component
			.find('.select-item')
			.at(1)
			.simulate('click', {
				stopPropagation
			})

		component
			.find('.input-item')
			.at(0)
			.simulate('click', {
				stopPropagation
			})

		component
			.find('.input-item')
			.at(1)
			.simulate('click', {
				stopPropagation
			})
		const tree = component.html()

		expect(tree).toMatchSnapshot()
		expect(stopPropagation).toBeCalledTimes(4)
	})

	test('NumericOption renders as expected with default requirement', () => {
		const component = renderer.create(<NumericOption numericChoice={{ requirement: '' }} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('NumericOption input calls focus on the element after a short delay', () => {
		jest.useFakeTimers()

		const component = renderer.create(<NumericOption numericChoice={{ requirement: '' }} />)

		const event = { target: { focus: jest.fn() }, stopPropagation: jest.fn() }
		component.root.findAllByType('input')[0].props.onFocus(event)

		jest.runAllTimers()

		expect(event.stopPropagation).toHaveBeenCalled()
		expect(event.target.focus).toHaveBeenCalledTimes(2)
	})
})
