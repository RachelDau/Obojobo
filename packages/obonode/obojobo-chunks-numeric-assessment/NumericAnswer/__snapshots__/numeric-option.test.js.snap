// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericOption NumericOption renders as expected with default requirement 1`] = `
<div
  className="is-type-exact"
>
  <label
    className="select requirement"
  >
    Answer Type
    <select
      className="select-item"
      name="requirement"
      onClick={[Function]}
    >
      <option>
        Exact answer
      </option>
      <option>
        Margin of error
      </option>
      <option>
        Within a range
      </option>
    </select>
  </label>
  <label
    className="input answer"
  >
    Answer
    <input
      autoComplete="off"
      className="input-item"
      contentEditable={false}
      name="answer"
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      value=""
    />
  </label>
</div>
`;

exports[`NumericOption NumericOption renders as expected with requirement of \`exact\` 1`] = `
<div
  className="is-type-exact"
>
  <label
    className="select requirement"
  >
    Answer Type
    <select
      className="select-item"
      name="requirement"
      onClick={[Function]}
      value="Exact answer"
    >
      <option>
        Exact answer
      </option>
      <option>
        Margin of error
      </option>
      <option>
        Within a range
      </option>
    </select>
  </label>
  <label
    className="input answer"
  >
    Answer
    <input
      autoComplete="off"
      className="input-item"
      contentEditable={false}
      name="answer"
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      value=""
    />
  </label>
</div>
`;

exports[`NumericOption NumericOption renders as expected with requirement of \`margin\` 1`] = `
<div
  className="is-type-margin"
>
  <label
    className="select requirement"
  >
    Answer Type
    <select
      className="select-item"
      name="requirement"
      onClick={[Function]}
      value="Margin of error"
    >
      <option>
        Exact answer
      </option>
      <option>
        Margin of error
      </option>
      <option>
        Within a range
      </option>
    </select>
  </label>
  <label
    className="input answer"
  >
    Answer
    <input
      autoComplete="off"
      className="input-item"
      contentEditable={false}
      name="answer"
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      value=""
    />
  </label>
  <label
    className="select margin-type"
  >
    Error Type
    <select
      className="select-item"
      name="margin-type"
      onClick={[Function]}
    >
      <option>
        Percent
      </option>
      <option>
        Absolute
      </option>
    </select>
  </label>
  <label
    className="input margin-value"
  >
    ± Error
    <input
      autoComplete="off"
      className="input-item"
      contentEditable={false}
      name="margin"
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      value=""
    />
  </label>
</div>
`;

exports[`NumericOption NumericOption renders as expected with requirement of \`range\` 1`] = `
<div
  className="is-type-range"
>
  <label
    className="select requirement"
  >
    Answer Type
    <select
      className="select-item"
      name="requirement"
      onClick={[Function]}
      value="Within a range"
    >
      <option>
        Exact answer
      </option>
      <option>
        Margin of error
      </option>
      <option>
        Within a range
      </option>
    </select>
  </label>
  <label
    className="input start"
  >
    Start
    <input
      autoComplete="off"
      className="input-item"
      contentEditable={false}
      name="start"
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      value=""
    />
  </label>
  <label
    className="input end"
  >
    End
    <input
      autoComplete="off"
      className="input-item"
      contentEditable={false}
      name="end"
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      value=""
    />
  </label>
</div>
`;

exports[`NumericOption NumericOption with requirement of \`exact\`, event.stopPropagation() is called when click 1`] = `"<div class=\\"is-type-exact\\"><label class=\\"select requirement\\">Answer Type<select class=\\"select-item\\" name=\\"requirement\\"><option>Exact answer</option><option>Margin of error</option><option>Within a range</option></select></label><label class=\\"input answer\\">Answer<input class=\\"input-item\\" name=\\"answer\\" contenteditable=\\"false\\" autocomplete=\\"off\\" value=\\"\\"></label></div>"`;

exports[`NumericOption NumericOption with requirement of \`margin\`, event.stopPropagation() is called when click 1`] = `"<div class=\\"is-type-margin\\"><label class=\\"select requirement\\">Answer Type<select class=\\"select-item\\" name=\\"requirement\\"><option>Exact answer</option><option>Margin of error</option><option>Within a range</option></select></label><label class=\\"input answer\\">Answer<input class=\\"input-item\\" name=\\"answer\\" contenteditable=\\"false\\" autocomplete=\\"off\\" value=\\"\\"></label><label class=\\"select margin-type\\">Error Type<select class=\\"select-item\\" name=\\"margin-type\\"><option>Percent</option><option>Absolute</option></select></label><label class=\\"input margin-value\\">± Error<input class=\\"input-item\\" name=\\"margin\\" contenteditable=\\"false\\" autocomplete=\\"off\\" value=\\"\\"></label></div>"`;

exports[`NumericOption NumericOption with requirement of \`margin\`, percent error, event.stopPropagation() is called when click 1`] = `"<div class=\\"is-type-margin\\"><label class=\\"select requirement\\">Answer Type<select class=\\"select-item\\" name=\\"requirement\\"><option>Exact answer</option><option>Margin of error</option><option>Within a range</option></select></label><label class=\\"input answer\\">Answer<input class=\\"input-item\\" name=\\"answer\\" contenteditable=\\"false\\" autocomplete=\\"off\\" value=\\"\\"></label><label class=\\"select margin-type\\">Error Type<select class=\\"select-item\\" name=\\"margin-type\\"><option>Percent</option><option>Absolute</option></select></label><label class=\\"input margin-value\\">% Error<input class=\\"input-item\\" name=\\"margin\\" contenteditable=\\"false\\" autocomplete=\\"off\\" value=\\"\\"></label></div>"`;

exports[`NumericOption NumericOption with requirement of \`range\`, event.stopPropagation() is called when click 1`] = `"<div class=\\"is-type-range\\"><label class=\\"select requirement\\">Answer Type<select class=\\"select-item\\" name=\\"requirement\\"><option>Exact answer</option><option>Margin of error</option><option>Within a range</option></select></label><label class=\\"input start\\">Start<input class=\\"input-item\\" name=\\"start\\" contenteditable=\\"false\\" autocomplete=\\"off\\" value=\\"\\"></label><label class=\\"input end\\">End<input class=\\"input-item\\" name=\\"end\\" contenteditable=\\"false\\" autocomplete=\\"off\\" value=\\"\\"></label></div>"`;
